### PR TITLE
DTSRD-3528.Extend the Get refdata/commondata/lov/categories/{categoryId} API to return the new values

### PR DESCRIPTION
**Pull request by SabinaHMCTS**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCategoriesIntegrationTest.java [hunk 1]
```diff
@@ -30,6 +33,56 @@ public class RetrieveCategoriesIntegrationTest extends CdAuthorizationEnabledInt
...
+
+        assertThat(response.getListOfCategory().get(2).getKey()).isEqualTo("PC1-01-74");
+        assertThat(response.getListOfCategory().get(2).getValueEn()).isEqualTo("Medical office holder");
+        assertThat(response.getListOfCategory().get(2).getExternalReference()).isEqualTo("74");
+        assertThat(response.getListOfCategory().get(2).getExternalReferenceType()).isEqualTo("MedicalRole");
+    }
+
+    @Test
+    @DisplayName("Retrieve categories without externalRefarence ")
+    @Description("test to show that Json igone did not pick up fields external_reference "
```
Comment (lukasz-wolski) [inline src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCategoriesIntegrationTest.java:65]: please clarify 'igone'. possibly a typo (Json ignore)?

File: src/integrationTest/java/uk/gov/hmcts/reform/cdapi/util/CommonDataApiClient.java [hunk 2]
```diff
@@ -82,6 +92,27 @@ private Object mapCaseFlagsByServiceIdResponse(ResponseEntity<Object> responseEn
 
     }
 
+    //to Test Json include only not empty vlues property
```
Comment (lukasz-wolski) [inline src/integrationTest/java/uk/gov/hmcts/reform/cdapi/util/CommonDataApiClient.java:95]: typo: should be values

File: src/main/resources/db/migration/V1_14__Alter_List_Of_Values.sql [hunk 3]
```diff
@@ -0,0 +1,10 @@
+
+ALTER TABLE List_Of_Values ADD COLUMN external_reference VARCHAR(200);
+ALTER TABLE List_Of_Values ADD COLUMN external_reference_type VARCHAR(200);
+
+
+ALTER TABLE list_of_values
```
Comment (lukasz-wolski) [inline src/main/resources/db/migration/V1_14__Alter_List_Of_Values.sql:6]: minor: readability - please decide on either lower case or capitalised case across migration file

**Review by lukasz-wolski — CHANGES_REQUESTED** [review]
please see comments inlined

File: src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCategoriesIntegrationTest.java (hunk 1 above)
Comment (SabinaHMCTS) [reply src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCategoriesIntegrationTest.java]: changed

File: src/integrationTest/java/uk/gov/hmcts/reform/cdapi/util/CommonDataApiClient.java (hunk 2 above)
Comment (SabinaHMCTS) [reply src/integrationTest/java/uk/gov/hmcts/reform/cdapi/util/CommonDataApiClient.java]: changed

File: src/main/resources/db/migration/V1_14__Alter_List_Of_Values.sql (hunk 3 above)
Comment (SabinaHMCTS) [reply src/main/resources/db/migration/V1_14__Alter_List_Of_Values.sql]: changed to upper

File: src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCategoriesIntegrationTest.java [hunk 4]
```diff
@@ -30,6 +33,56 @@ public class RetrieveCategoriesIntegrationTest extends CdAuthorizationEnabledInt
 
     private static final String path = "/lov/categories/{category-id}";
 
+    @Test
+    @DisplayName("Retrieve categories with externalRefarence ")
```
Comment (mikebrownccd) [inline src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCategoriesIntegrationTest.java:37]: Typo, "externalRefarence" should be "externalReference"

File: src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCategoriesIntegrationTest.java [hunk 5]
```diff
@@ -30,6 +33,56 @@ public class RetrieveCategoriesIntegrationTest extends CdAuthorizationEnabledInt
...
+        assertThat(response.getListOfCategory().get(1).getExternalReferenceType()).isEqualTo("JudicialRole");
+
+        assertThat(response.getListOfCategory().get(2).getKey()).isEqualTo("PC1-01-74");
+        assertThat(response.getListOfCategory().get(2).getValueEn()).isEqualTo("Medical office holder");
+        assertThat(response.getListOfCategory().get(2).getExternalReference()).isEqualTo("74");
+        assertThat(response.getListOfCategory().get(2).getExternalReferenceType()).isEqualTo("MedicalRole");
+    }
+
+    @Test
+    @DisplayName("Retrieve categories without externalRefarence ")
```
Comment (mikebrownccd) [inline src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCategoriesIntegrationTest.java:64]: Typo, "externalRefarence" should be "externalReference"

File: src/main/java/uk/gov/hmcts/reform/cdapi/domain/ListOfValueDto.java [hunk 6]
```diff
@@ -36,8 +36,10 @@ public class ListOfValueDto {
     private String parentKey;
 
     private String active;
```
Comment (mikebrownccd) [inline src/main/java/uk/gov/hmcts/reform/cdapi/domain/ListOfValueDto.java:38]: strange that active doesn't have a @Column annotation

**Review by mikebrownccd — COMMENTED** [review]
I have left some comments

File: src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCategoriesIntegrationTest.java (hunk 5 above)
Comment (SabinaHMCTS) [reply src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCategoriesIntegrationTest.java]: corrected

File: src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCategoriesIntegrationTest.java (hunk 4 above)
Comment (SabinaHMCTS) [reply src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCategoriesIntegrationTest.java]: corrected

**Review by lukasz-wolski — APPROVED** [review]

**Pull request merged**